vms/platformvm/validators/manager.go

```diff
@@ -170,56 +171,105 @@ func (m *manager) GetValidatorSet(ctx context.Context, height uint64, subnetID i
 		return nil, ErrMissingValidator
 	}
 
-	currentSubnetValidatorList := currentSubnetValidators.List()
-	vdrSet := make(map[ids.NodeID]*validators.GetValidatorOutput, len(currentSubnetValidatorList))
-	for _, vdr := range currentSubnetValidatorList {
-		primaryVdr, ok := currentPrimaryNetworkValidators.Get(vdr.NodeID)
+	currentSubnetValidatorList := currentSubnetValidatorSet.List()
+	currentSubnetValidators := make([]*maskedValidator, 0, len(currentSubnetValidatorList))
+	currentSubnetValidatorIndices := make(map[ids.NodeID]int)
+	for i, validator := range currentSubnetValidatorList {
+		primaryVdr, ok := currentPrimaryNetworkValidators.Get(validator.NodeID)
 		if !ok {
 			// This should never happen
-			return nil, fmt.Errorf("%w: %s", ErrMissingValidator, vdr.NodeID)
-		}
-		vdrSet[vdr.NodeID] = &validators.GetValidatorOutput{
-			NodeID:    vdr.NodeID,
-			PublicKey: primaryVdr.PublicKey,
-			Weight:    vdr.Weight,
+			return nil, fmt.Errorf("%w: %s", ErrMissingValidator, validator.NodeID)
 		}
+
+		currentSubnetValidators = append(currentSubnetValidators, &maskedValidator{
+			GetValidatorOutput: &validators.GetValidatorOutput{
+				NodeID:    validator.NodeID,
+				PublicKey: primaryVdr.PublicKey,
+				Weight:    validator.Weight,
+			},
+			masked: false,
+		})
+		currentSubnetValidatorIndices[validator.NodeID] = i
 	}
 
+	validatorsFromDiffs := make([]*maskedValidator, 0)
+	validatorsFromDiffsIndices := make(map[ids.NodeID]int)
```

Comment:
in current code version, `validatorsFromDiffsIndices` is used only in `applyValidatorDiffs`. Move in `applyValidatorDiffs`